### PR TITLE
using raw event payload in signature verification, json parse payload for recipient verification

### PR DESCRIPTION
**Pull request by abhupadh**
## Description

This adds support to parse a JSON string with field values that may contain large numbers outside the javascript standard numeric range. As JavaScript uses floating-point numbers the accuracy is only assured for integers between: -9007199254740992 (-2^53) and 9007199254740992 (2^53). Any number outside this range gets converted to its nearest representation of the JS number.

## Motivation and Context

A user reported issue with normal JSON.parse parsing his payload with large number fields to a different value due to which his signature verification was failing. 

Relevant JSON.parse limitation issue reported on [stackoverflow](https://stackoverflow.com/questions/10631494/json-parse-parses-converts-big-numbers-incorrectly) 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -2,7 +2,10 @@
   "name": "@adobe/aio-lib-events",
   "version": "1.1.3",
   "description": "Adobe I/O Events",
-  "repository": "https://github.com/adobe/aio-lib-events",
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/adobe/aio-lib-events.git"
```
Comment (shikhartanwar) [inline package.json:7]: Is this change intentional?

File: src/helpers.js [hunk 2]
```diff
@@ -95,10 +96,13 @@ function getProperPayload (event) {
...
     if (isBase64Encoded(event)) {
-      decodedJsonPayload = JSON.parse(Buffer.from(event, 'base64').toString('utf-8'))
+      // using JSONbig.parse to handle payloads which may contain big integers.
+      // cf https://www.irt.org/script/1031.htm
+      // using JSON.parse, there will be data loss as these big numbers outside the range gets transformed to nearest Javascript number
+      decodedJsonPayload = JSONbig.parse(Buffer.from(event, 'base64').toString('utf-8'))
     } else {
       // parsing for non-encoded json payloads (e.g. custom events)
-      decodedJsonPayload = JSON.parse(event)
+      decodedJsonPayload = JSONbig.parse(event)
```
Comment (shikhartanwar) [inline src/helpers.js:105]: Does this guarantee ordering of keys because even that could cause the signature verification to fail? 
I would suggest working with the raw payload instead (without any JSON parsing). WDYT?

File: package.json (hunk 1 above)
Comment (abhupadh) [reply package.json:7]: got generated automatically when did `npm init` locally.

File: src/helpers.js (hunk 2 above)
Comment (abhupadh) [reply src/helpers.js:105]: It's built on top of the `JSON.parse` and I tested multiple times which parses payload just fine. Working with raw payload wouldn't work as we need to fetch the `recipient_client_id` from the payload, which I am not sure if we can get unless we parse the json we receive

Comment (shikhartanwar) [reply src/helpers.js:105]: We can verify the signature first, then verify the `recipient_client_id` by parsing the json payload so that the order of keys does not matter anymore. WDYT?

File: package.json (hunk 1 above)
Comment (shikhartanwar) [reply package.json:7]: ok

File: src/helpers.js (hunk 2 above)
Comment (abhupadh) [reply src/helpers.js:105]: what we can do is use the parsed payload to verify the `recipient_client_id` and then use the raw payload (which I beleive won't change at this point here) for the signature verification. Else the whole verification sequence has to be changed in the code. WDYT?

Comment (abhupadh) [reply src/helpers.js:105]: Okay. I will verify the above and change accordingly.

File: test/helpers.test.js [hunk 3]
```diff
@@ -61,11 +61,6 @@ describe('Proper Payload Test', () => {
     const encodedValidPayload = mock.data.testEncodedPayload.event
     const decodedJsonPayload = mock.data.testEvent.event
     const res = await helpers.getProperPayload(encodedValidPayload)
-    expect(JSON.stringify(res)).toEqual(decodedJsonPayload)
-  })
-  it('test invalid payload returns error', async () => {
-    const encodedInvalidPayload = mock.data.testEncodedInvalidPayload.event
-    const res = await helpers.getProperPayload(encodedInvalidPayload)
-    expect(res.error.statusCode).toBe(400)
+    expect(res).toEqual(decodedJsonPayload)
```
Comment (abhupadh) [inline test/helpers.test.js:64]: fixing test error coming in previous commit

File: src/signatureUtils.js [hunk 4]
```diff
@@ -227,12 +227,12 @@ async function createCryptoPublicKey (publicKey) {
 /**
  * Checks if the recipient client id is the valid target recipient of the event payload
  *
- * @param {*} decodedJsonPayload a valid json event payload used in I/O Events signing
+ * @param {*} parsedJsonPayload a valid deserialized json event payload object used in I/O Events signing
```
Comment (shikhartanwar) [inline src/signatureUtils.js:230]: We can simply say.
```suggestion
 * @param {*} parsedJsonPayload deserialized json event payload
```

**Comment by shikhartanwar** [comment]
@abhupadh Can we please update the description and title of this PR since we are not using JSONbig library anymore?

**Review by shikhartanwar — APPROVED** [review]
